app/Jobs/DeleteQueryserviceNamespaceJob.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Jobs;
+
+use App\QueryserviceNamespace;
+use App\Http\Curl\CurlRequest;
+use App\Http\Curl\HttpRequest;
+use App\Wiki;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+
+class DeleteQueryserviceNamespaceJob extends Job implements ShouldBeUnique
+{
+    private $wikiId;
+    private $request;
+
+    /**
+     * @return void
+     */
+    public function __construct( $wikiId, HttpRequest $request = null)
+    {
+        $this->wikiId = $wikiId;
+        $this->request = $request ?? new CurlRequest();
+    }
+
+    /**
+     * The unique ID of the job.
+     *
+     * @return string
+     */
+    public function uniqueId()
+    {
+        return strval($this->wikiId);
+    }
+
+    /**
+     * @return void
+     */
+    public function handle()
+    {
+        $wiki = Wiki::withTrashed()->where('id', $this->wikiId)->first();
+
+        if( !$wiki ) {
+            $this->fail( new \RuntimeException("Wiki {$this->wikiId} not found.") );
+            return;
+        }
```

Comment:
The `$wiki` doesn’t seem to be used for anything, as far as I can tell… is the error that’s being thrown important? I assume if the wiki doesn’t exist, we would still get an error a few lines below when `$qsNamespace` can’t be found.

Reply:
yep true, lets remove it